src/Domain/LeanCode.DomainModels/Ulids/Ulid.cs

```diff
@@ -0,0 +1,884 @@
+        span[19] = Base32Text[((randomness5 & 7) << 2) | ((randomness6 & 192) >> 6)];
+        span[20] = Base32Text[(randomness6 & 62) >> 1];
+        span[21] = Base32Text[((randomness6 & 1) << 4) | ((randomness7 & 240) >> 4)];
+        span[22] = Base32Text[((randomness7 & 15) << 1) | ((randomness8 & 128) >> 7)];
+        span[23] = Base32Text[(randomness8 & 124) >> 2];
+        span[24] = Base32Text[((randomness8 & 3) << 3) | ((randomness9 & 224) >> 5)];
+
+        return true;
+    }
+
+    public override string ToString()
+    {
+        return string.Create(26, this, (span, ulid) => ulid.TryWriteStringify(span));
+    }
+
+    public string ToString(string? format, IFormatProvider? formatProvider) => ToString();
+
+    public bool TryFormat(
+        Span<char> destination,
+        out int charsWritten,
+        ReadOnlySpan<char> format,
+        IFormatProvider? provider
+    )
+    {
+        if (TryWriteStringify(destination))
+        {
+            charsWritten = 26;
+            return true;
+        }
+        else
+        {
+            charsWritten = 0;
+            return false;
+        }
+    }
+
+    // Comparable/Equatable
+
+    public override unsafe int GetHashCode()
+    {
+        // Simply XOR, same algorithm of Guid.GetHashCode
+        fixed (void* p = &this.timestamp0)
+        {
+            var a = (int*)p;
+            return (*a) ^ *(a + 1) ^ *(a + 2) ^ *(a + 3);
+        }
+    }
+
+    public unsafe bool Equals(Ulid other)
+    {
+        // readonly struct can not use Unsafe.As...
+        fixed (byte* a = &this.timestamp0)
+        {
+            byte* b = &other.timestamp0;
+
+            {
+                var x = *(ulong*)a;
+                var y = *(ulong*)b;
+                if (x != y)
+                {
+                    return false;
+                }
+            }
+            {
+                var x = *(ulong*)(a + 8);
+                var y = *(ulong*)(b + 8);
+                if (x != y)
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+    }
+
+    public override bool Equals(object? obj)
+    {
+        return obj is Ulid other && this.Equals(other);
+    }
+
+    public static bool operator ==(Ulid a, Ulid b)
+    {
+        return a.Equals(b);
+    }
+
+    public static bool operator !=(Ulid a, Ulid b)
+    {
+        return !a.Equals(b);
+    }
+
+    public static bool operator <(Ulid a, Ulid b) => a.CompareTo(b) < 0;
+
+    public static bool operator <=(Ulid a, Ulid b) => a.CompareTo(b) <= 0;
+
+    public static bool operator >(Ulid a, Ulid b) => a.CompareTo(b) > 0;
+
+    public static bool operator >=(Ulid a, Ulid b) => a.CompareTo(b) >= 0;
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    private static int GetResult(byte me, byte them) => me < them ? -1 : 1;
+
+    public int CompareTo(Ulid other)
+    {
+        if (this.timestamp0 != other.timestamp0)
```

Comment:
And here, reinterpret memory as `ROS<byte>` and run compare on that with `SequenceCompareTo`. Might be a bit slower, but I'd be okay with that.